src/char/int_rodex.c

```diff
@@ -0,0 +1,629 @@
+			"FROM `rodex_mail` WHERE `expire_date` > '%d' AND `receiver_id` = '%d'"
+			"ORDER BY `id` ASC", (int)time(NULL), char_id)
+			) {
+			SqlStmt_ShowDebug(stmt);
+			SQL->StmtFree(stmt);
+			return -1;
+		}
+		break;
+
+	case RODEX_OPENTYPE_ACCOUNT:
+		// Prepares the statement to retrieve first page of account box
+		if (SQL_ERROR == SQL->StmtPrepare(stmt,
+			"SELECT `id`, `sender_name`, `sender_id`, `receiver_name`, `receiver_id`, `receiver_accountid`,"
+			"`title`, `body`, `zeny`, `type`, `is_readed`, `send_date`, `expire_date`, `weight`"
+			"FROM `rodex_mail` WHERE "
+			"`expire_date` > '%d' AND `receiver_accountid` = '%d'"
+			"ORDER BY `id` ASC", (int)time(NULL), account_id)
+			) {
+			SqlStmt_ShowDebug(stmt);
+			SQL->StmtFree(stmt);
+			return -1;
+		}
+		break;
+
+	case RODEX_OPENTYPE_RETURN:
+		// Prepares the statement to retrieve first page of returned box
+		if (SQL_ERROR == SQL->StmtPrepare(stmt,
+			"SELECT `id`, `sender_name`, `sender_id`, `receiver_name`, `receiver_id`, `receiver_accountid`,"
+			"`title`, `body`, `zeny`, `type`, `is_readed`, `send_date`, `expire_date`, `weight`"
+			"FROM `rodex_mail` WHERE (`sender_id` = '%d' AND `expire_date` <= '%d' AND `send_date` + '%d' > '%d')"
+			"ORDER BY `id` ASC", char_id, (int)time(NULL), 2 * RODEX_EXPIRE, (int)time(NULL))
+			) {
+			SqlStmt_ShowDebug(stmt);
+			SQL->StmtFree(stmt);
+			return -1;
+		}
+		break;
+	}
+
+	if (SQL_ERROR == SQL->StmtExecute(stmt)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 0, SQLDT_INT64, &msg.id, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 1, SQLDT_STRING, &msg.sender_name, sizeof(msg.sender_name), NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 2, SQLDT_INT, &msg.sender_id, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 3, SQLDT_STRING, &msg.receiver_name, sizeof(msg.receiver_name), NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 4, SQLDT_INT, &msg.receiver_id, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 5, SQLDT_INT, &msg.receiver_accountid, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 6, SQLDT_STRING, &msg.title, sizeof(msg.title), NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 7, SQLDT_STRING, &msg.body, sizeof(msg.body), NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 8, SQLDT_INT, &msg.zeny, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 9, SQLDT_UINT8, &msg.type, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 10, SQLDT_INT8, &msg.is_readed, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 11, SQLDT_INT, &msg.send_date, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 12, SQLDT_INT, &msg.expire_date, 0, NULL, NULL)
+		|| SQL_ERROR == SQL->StmtBindColumn(stmt, 13, SQLDT_INT, &msg.weight, 0, NULL, NULL)
+		) {
+		SqlStmt_ShowDebug(stmt);
+		SQL->StmtFree(stmt);
+		return -1;
+	}
+
+	// Read mails
+	while (SQL_SUCCESS == SQL->StmtNextRow(stmt)) {
```

Comment:
for future reference, I'm splitting the packets when sending to map (on `mapif_rodex_sendinbox`)